babi/screen.py

```diff
@@ -131,12 +131,19 @@ def _draw_header(self) -> None:
             filename += ' *'
         if len(self.files) > 1:
             files = f'[{self.i + 1}/{len(self.files)}] '
-            version_width = len(VERSION_STR) + 2 + len(files)
         else:
             files = ''
-            version_width = len(VERSION_STR) + 2
-        centered = filename.center(self.margin.cols)[version_width:]
-        s = f' {VERSION_STR} {files}{centered}{files}'
+
+        before, after = f' {VERSION_STR} {files}', f'{files}'
+        room_for_filename = self.margin.cols - 2 * len(before)
+        if len(filename) > room_for_filename:
+            truncate = '... '
```

Comment:
the rest of the project utilizes the ellipsis character -- this also makes truncation preserve more characters